### DISCUSSION TITLE
Game Backup Monitor v1.3.2

### DISCUSSION THREAD

**Discussion by MikeMaximus** [MikeMaximus/gbm#285] [General]
***New in 1.3.2***

All Platforms:

- Fixed an issue that could cause imports from the web to stop functioning after a failed download.
- Fixed an issue that caused recent changes to be lost if GBM was unexpectedly closed while working in certain windows, like the Game Manager.
- Fixed a crash that occurred when the "Configuration Links" window is opened on the Game Manager while adding a new configuration.
- Fixed various issues when importing backup files:
	- Backup files created by v1.2.8 can now be imported without errors.
	- Fixed the broken 7z file filter when importing backup files.	
	- Fixed an issue that could cause ManifestID constraint errors.
	- Fixed an issue could cause differential backup imports to always fail.
- Changes to folder and file naming:
	- Removed the "Use Game ID for folder and file names" option from Settings.
	- Configurations using duplicate names will now use a more descriptive name format:
		- Game Name [Game ID] Ex. Metro Exodus [8ac11c11-8d18-471b-8f0d-aa154dc77f0e]	
	- The length of folder and file names is now more strict to minimize issues with the Windows max path limitation.
		- Only the first 64 characters of a game name will be used when creating folder and file names.
	- Modifying the name of a configuration will no longer automatically rename existing backup folders and files.
	- The above changes are only applied when creating new backup files.
	- I understand the recent naming changes may be frustrating to some users.  Please leave any feedback in the GitHub issues or discussions section.
- Improvements to the Game Manager:
	- Added a "Copy" feature.
		- This will make a copy of the currently selected game configuration(s).
		- The Core Configuration, Game Information, Tags, Processes and Configuration Links are included in the copy.
	- Added buttons to quickly open the current "Game Path" or "Save Path".
	- Some game configuration options have been renamed for clarification.
		- The "Monitor this game" option has been renamed to "Allow monitoring".
		- The "Monitor only" option has been renamed to "No backup when game ends".
	- The "Backup limit" and "Full backup interval" fields will no longer reset to 0 in the Game Manager if toggled off during an edit.
- Improvements to Importing Game Configurations:
	- The list will now try to retain the last scroll position after being refreshed.
	- You can now "Ignore" (or "Unignore") any configurations on the import window using the right-click menu.
		- This allows you to hide any configuration(s) that you never want to add or update during an import.
		- The ignored configurations are saved and persist between sessions.
		- You can show any hidden configurations by unticking the "Hide Ignored" checkbox.
	- Added new icons to indicate auto-detected and ignored configurations.
- General Interface improvements:
	- Added an "Open Backup Folder" option to the File menu (Main window and system tray).
	- You can now filter by "Tag" specifically in some search fields by using a hashtag in the search term. Ex. #Steam
		- In prior versions you didn't need to use a hashtag in the search term to filter by tag, but the results included both name and tag matches.
		- This feature is supported on the Main window, the Game Manager, and the Import window.		 		
	- All windows can now be closed and/or canceled out of edit mode by using the "Escape" key.
	- More windows will now ask for confirmation before closing if changes aren't yet saved.
	- Double-clicking a game in the main window list will now open it in the Game Manager.
	- You can now use the "Enter" key to confirm the "Ludusavi Options" window.	
- General performance improvements:
	- Some features that caused the interface to become unresponsive now run on a seperate thread.
	- Optimized various database queries.
			
The entire version history of GBM releases is available at http://mikemaximus.github.io/gbm-web/versionhistory.html

<hr /><em>This discussion was created from the release <a href='https://github.com/MikeMaximus/gbm/releases/tag/v1.3.2'>Game Backup Monitor v1.3.2</a>.</em>

**Comment by MikeMaximus** [comment]
I hope this latest release is working well for everyone.  I have yet to receive a report of any major bugs or issues.

I'm busy this time of year, but will likely start work on the next version soon. v1.3.3 should include more UI improvements, some global settings for backups, and possibly a [Flatpak](https://www.flatpak.org/) release for Linux so GBM is easier to use with the Steam Deck.

**Reply by DustinVG** [reply]
It's been working fine as far as I can tell!

I literally _just_ got a Steam Deck, so a Flatpak release would be phenomenally timed.

**Comment by MikeMaximus** [comment]
Unfortunately it's looking like a Flatpak release won't be happening soon.  😞 

I've run into a Mono related issue with GBM in a Flatpak that I may not be able to fix.  I might take another crack at it, but I think my time might be better spent working on the app itself.

**Reply by DustinVG** [reply]
Unfortunate, but understandable.

Do you have any advice for alternative methods on running it on Deck if it doesn't work out?

**Reply by MikeMaximus** [reply]
I don't have a deck myself and probably won't for the foreseeable future.  But from what I've heard you can install any app and their prerequisites manually.  But they will get wiped each time there's a system update and need to be reinstalled, using flatpak based apps prevents that.

I'm actually not even sure anymore which distro Steam OS is based on nowadays.  If it's Arch you may be able to use https://aur.archlinux.org/packages/gbm to install it, if it's still on Debian using the release deb package should work too.

**Reply by DustinVG** [reply]
It used to be Debian but now seems to be Arch.
